eng/common/pipelines/templates/steps/set-default-branch.yml

```diff
@@ -4,7 +4,12 @@ parameters:
 steps:
 - pwsh: |
     $setDefaultBranch = (git remote show ${{ parameters.RemoteRepo }} | Out-String) -replace "(?ms).*HEAD branch: (\w+).*", '$1'
+    if ($LASTEXITCODE -ne 0) {
+      Write-Host "Not able to fetch the default branch from git command. Set to master."
+      $setDefaultBranch = 'master'
+    }
     Write-Host "Setting DefaultBranch=$setDefaultBranch"
     echo "##vso[task.setvariable variable=DefaultBranch]$setDefaultBranch"
   displayName: "Setup Default Branch"
   workingDirectory: ${{ parameters.workingDirectory }}
+  continueOnError: true 
```

Comment:
This will end up turning the pipeline yellow which we should try to avoid. Please do a test build on a private pipeline with this change and lets see how we can make this silent (i.e. no warnings/errors) only the write-host message.

Reply:
Will have a pipeline on my private fork repo

Reply:
https://dev.azure.com/azure-sdk/playground/_build/results?buildId=845015&view=logs&j=38d918fc-df5a-5870-2b80-6c17fdb13baf&t=489baed9-6f06-5266-9d0d-079a130738b4

Reply:
Ok that looks better.